### ISSUE TITLE
Add TCP Support

### ISSUE THREAD

**Issue by talevy** [logstash-plugins/logstash-output-gelf#1]
migrated from: https://logstash.jira.com/browse/LOGSTASH-780

Graylog 0.10 starts supporting TCP transport for gelf.
Since this enables safer transport than UDP, I suggest extending the gelf input/output with a TCP option.

**Comment by florian-asche** [comment]
+1

Gelf-rb starts support of gelf tcp. Please suggest extending!

**Comment by runningman84** [comment]
this looks like a simple change:

```
# old
# @gelf ||= GELF::Notifier.new(@host, @port, @chunksize)
# new
@gelf ||= GELF::Notifier.new(@host, @port, "LAN", { :protocol => GELF::Protocol::TCP })
```

**Comment by rcord02** [comment]
+1
would be awesome if this will be availible in the next release!

**Comment by danlsgiga** [comment]
+1... This is a must for our environment. Please bring this new feature in the next release.
Also, @runningman84 I've tested the simple change you suggested but it does not work.

**Comment by danlsgiga** [comment]
FYI...
https://github.com/Graylog2/gelf-rb/pull/21

**Comment by ghost** [comment]
Are there any plans to work on this enhancement?  TCP is a must for deploying a scalable Graylog cluster in AWS, since AWS elastic load balancers do not support UDP

**Comment by alpha-centauri** [comment]
Still no TCP in 2017?! So TCP support in Graylog2/gelf-rb has been developed all in vain, indeed.

**Comment by runningman84** [comment]
this can be closed due to https://github.com/logstash-plugins/logstash-output-gelf/issues/19

**Comment by Moeser** [comment]
For those landing on this from Google, the TCP option for logstash's gelf output plugin has been available since v3.0.0 of the plugin.  There's a (currently undocumented) option you can add to the output plugin `protocol => "TCP"` that switches it to TCP instead of UDP.

**Comment by davesave** [comment]
I hope someday someone will fine the time and update 'docs/index.asciidoc' with this hidden protocol => "tcp" option.
